src/views/search/controls/SearchControls.tsx

```diff
@@ -42,10 +42,16 @@ export const SearchControls = ({ hasQuery }: Props) => {
   const filterOptions: Array<{ value: string; text: string }> = [];
 
   if (filters.isEnabled) {
-    filterOptions.push({
-      text: "All sources",
-      value: "",
-    });
+    if (filters.allSources) {
+      filterOptions.push({
+        text: "All sources",
+        value: "",
+      });
+    }
+
+    if (!filters.allSources && filterValue === "") {
+      setFilterValue(filters.sources[0].value)
```

Comment:
How does this code affect the UX? Can we note the intended effect in a code comment?